common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -409,6 +461,100 @@ bool mon_wlan_hal_dwpal::sta_link_measurements_11k_request(const std::string &st
     return true;
 }
 
+bool mon_wlan_hal_dwpal::channel_scan_trigger(int dwell_time_msec,
+                                              const std::vector<unsigned int> &channel_pool)
+{
+    LOG(DEBUG) << __func__ << " received on interface=" << m_radio_info.iface_name;
+
+    //build scan parameters
+    ScanParams channel_scan_params = {0};
+    sScanCfgParams org_fg, new_fg;   //foreground scan param
+    sScanCfgParamsBG org_bg, new_bg; //background scan param
+
+    // get original scan params
+    if (!dwpal_get_scan_params_fg(org_fg) || !dwpal_get_scan_params_bg(org_bg)) {
+        LOG(ERROR) << "Failed getting original scan parameters";
+        return false;
+    }
+
+    // prepare new scan params with changed dwell time
+    memcpy_s(&new_fg, sizeof(new_fg), &org_fg, sizeof(org_fg));
+    memcpy_s(&new_bg, sizeof(new_bg), &org_bg, sizeof(org_bg));
+    new_fg.passive = dwell_time_msec;
+    new_fg.active  = dwell_time_msec;
+    new_bg.passive = dwell_time_msec;
+    new_bg.active  = dwell_time_msec;
+
+    // set new scan params & get newly set values for validation
+    if (!dwpal_set_scan_params_fg(new_fg) || !dwpal_get_scan_params_fg(new_fg)) {
+        LOG(ERROR) << "Failed setting new values, restoring original scan parameters";
+        dwpal_set_scan_params_fg(org_fg);
+        dwpal_set_scan_params_bg(org_bg);
+        return false;
+    }
+    if (!dwpal_set_scan_params_bg(new_bg) || !dwpal_get_scan_params_bg(new_bg)) {
```

Comment:
Why do you need to get what you just set?

Reply:
For validation, moved the check to the validation's "if" statement